app/models/reports/schema_v2.rb

```diff
@@ -0,0 +1,228 @@
+    private def cumulative_assigned_patients_query_v2(region)
+      FacilityState.for_region(region)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("cumulative_assigned_patients::int")
+        .to_h(&period_hash)
+        .tap { |hsh| hsh.default = 0 }
+    end
+
+    # Returns registration counts per region / period
+    memoize def monthly_registrations
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = registered_patients_query_v2(region).slice(*periods.entries)
+      }
+    end
+
+    def registered_patients_query_v2(region)
+      return {} if earliest_patient_recorded_at_period[region.slug].nil?
+      FacilityState.for_region(region)
+        .where("month_date >= ?", earliest_patient_recorded_at_period[region.slug].to_date)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("monthly_registrations::int")
+        .to_h(&period_hash)
+    end
+
+    memoize def cumulative_registrations
+      regions.each_with_object({}) { |region, result|
+        result[region.slug] = cumulative_registered_patients_query_v2(region).slice(*periods.entries)
+      }
+    end
+
+    private def cumulative_registered_patients_query_v2(region)
+      return {} if earliest_patient_recorded_at_period[region.slug].nil?
+      FacilityState.for_region(region)
+        .where("month_date >= ?", earliest_patient_recorded_at_period[region.slug].to_date)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("cumulative_registrations::int")
+        .to_h(&period_hash)
+        .tap { |hsh| hsh.default = 0 }
+    end
+
+    memoize def ltfu
+      regions.each_with_object({}) { |region, hsh| hsh[region.slug] = ltfu_query_v2(region) }
+    end
+
+    private def ltfu_query_v2(region)
+      FacilityState.for_region(region)
+        .order(:month_date)
+        .group(:month_date)
+        .sum("lost_to_follow_up::int")
+        .to_h(&period_hash)
+        .tap { |hsh| hsh.default = 0 }
+    end
+
+    memoize def ltfu_rates
+      region_period_cached_query(__method__) do |entry|
+        slug, period = entry.slug, entry.period
+        percentage(ltfu[slug][period], cumulative_assigned_patients[slug][period])
+      end
+    end
+
+    memoize def controlled
+      regions.each_with_object({}).each do |region, hsh|
+        if earliest_patient_recorded_at[region.slug].nil?
+          hsh[region.slug] = Hash.new(0)
+          next
+        end
+        hsh[region.slug] = control_rate_query_v2.controlled_counts(region, range: active_range(region))
+      end
+    end
+
+    memoize def uncontrolled
+      regions.each_with_object({}).each do |region, hsh|
+        if earliest_patient_recorded_at[region.slug].nil?
+          hsh[region.slug] = Hash.new(0)
+          next
+        end
+        hsh[region.slug] = control_rate_query_v2.uncontrolled_counts(region, range: active_range(region))
+      end
+    end
```

Comment:
These need to be converted to use FacilityState ...future story here https://app.clubhouse.io/simpledotorg/story/4310/grab-as-much-repository-schema-v2-data-w-one-query-as-possible